xwiki-platform-core/xwiki-platform-link/src/main/java/org/xwiki/link/LinkStore.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.link;
+
+import java.util.Set;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Allow accessing the links extracted from various entities.
+ * 
+ * @version $Id$
+ * @since 14.8RC1
+ */
+@Role
+@Unstable
+public interface LinkStore
+{
+    // TODO: also introduce an API (#getLinks) to get all types of links and not only entities
```

Comment:
You're planning a future improvment ticket for this?

Reply:
Yes. No short term plan to work on it, but I will create issues for all the TODO, but I prefer to keep the TODO to help whoever will work on those new features in the future.

Reply:
Sounds good, wanted to be sure it was not leftover that you were planning to have before merging.